bugfix: redefine reader association policy

chrox opened the pull request:
Users can still use default reader for specific type. The "Remember this file" option will add exception to the predefined reader preference. And exception for a document can be made with the "Alt+C" command after opening the doc. This patch fixed the bug reported on issue #480 by @dracodoc.

chrox committed:
bugfix: redefine reader association policy
Users can still use default reader for specific type. The "Remember
this file" option will add exception to the predefined reader
preference. And exception for a document can be made with the "Alt+C"
command after opening the doc. This patch fixed the bug reported on
issue #480 by @dracodoc.

tigran123 commented:
Thank you! :)

tigran123 commented:
I have just tested it and it breaks PDF support entirely, i.e. after opening a DjVu file (with standard reader) and remembering the association "for this file type (T)" and then closing the file and trying to open a PDF file (with no prior history) it does not open it but refreshes the screen and remains in filechooser.

Switching back to master branch and doing the same thing works fine.

chrox committed:
bugfix: reader chooser dialog is shown properly
When there are reader preferences table but preference for the file type
of which the file is to be opened is not available, reader chooser
should show up.
I changed `last_reader` and `default_reader` to `reader_association` and
`reader_preferences` respectively for clearness.

chrox commented:
Patch 2126166 fixed the bug mentioned above.

tigran123 commented:
Thanks, it works now.
